server/src/main/java/org/elasticsearch/index/mapper/FieldMapper.java

```diff
@@ -63,15 +60,20 @@
 public abstract class FieldMapper extends Mapper {
     private static final Logger logger = LogManager.getLogger(FieldMapper.class);
 
-    public static final Setting<Boolean> IGNORE_MALFORMED_SETTING = Setting.boolSetting("index.mapping.ignore_malformed", settings -> {
-        if (IndexSettings.MODE.get(settings) == IndexMode.LOGSDB
-            && IndexMetadata.SETTING_INDEX_VERSION_CREATED.get(settings).onOrAfter(IndexVersions.ENABLE_IGNORE_MALFORMED_LOGSDB)) {
-            return "true";
-        } else {
-            return "false";
-        }
-    }, Property.IndexScope, Property.ServerlessPublic);
+    public static final Setting<Integer> IGNORE_ABOVE_SETTING = Setting.intSetting(
```

Comment:
Why is this added to `FieldMapper`? It only applies to a handful of mappers, iiuc.

Reply:
It is where `ignore_malformed` is...which looks a good place for me and despite `ignore_malformed` being for many more types is still not valid for all of them. Moreover `FieldMapper` is already imported by `WildcardFieldMapper` (which BTW is in xpack), and is in the same package of `KeywordFieldMapper` and `FlattenedFieldMapper`...which means also no additional import (and as such no additional dependency) is required. I don't see a better common class...the other option would be `Mapper`...but that I guess would trigger the same question. Any suggestion?

Reply:
It may be worth adding an intermediate class for such functionality, e.g. `BaseTextProcessingMapper` or something like that. Another one would be to just add it to `KeywordFieldMapper` and reference it elsewhere, since most other mappers use it. Not ideal, but seems preferable to `FieldMapper` imho.

Reply:
I'm still against polluting `FieldMapper` for functionality belonging to a small subset of subclasses. You can ask @martijnvg to approve this.

Reply:
Ignore malformed is a different feature than ignore above. Te ignore above feature only exists for a few field mappers.  We don't have a common base class for these field mappers. I think we should just move the setting to  `KeywordFieldMapper`.

Reply:
Or maybe better just move the setting to `IndexSettings`?